docs/overview.mdx

```diff
@@ -37,6 +37,64 @@ Casdoor serves both the web UI and the login requests from the application users
 
 9. Casdoor supports mainstream databases: MySQL, PostgreSQL, SQL Server etc,  and support the extension of new database with plugins.
 
+## How it works:
+
+### Step 0 (Pre-knowledge)
+1. `Casdoor`'s authorization process is built upon the OAuth 2.0 protocol, therefore it's highly recommended to take a brief look at how indeed does OAuth 2.0 works. An [introduction](https://www.digitalocean.com/community/tutorials/an-introduction-to-oauth-2) of OAuth 2.0.
+
+![image](https://user-images.githubusercontent.com/46557895/132336955-e25ab988-64b3-4087-b332-d9e724322acd.png)
+
+### Step 1 (Authorization Request) 
+
+Your `Application` (could be a website or whatever) should compose an `URL` in this format `endpoint/login/oauth/authorize?client_id=xxx&response_type=code&redirect_uri=xxx&scope=read&state=xxx`. In the `URL` replace `endpoint` with your `Casdoor`'s host URL, and replace `xxx` with your own info.
+
+**Hints**:  
+How to fill out the `xxx` parts?  
+
++ For `client_id`: you can find this under each single `Application`  
+
++ For `redirect_uri`: you should set this to your own `Application`'s callback URL, with this info, `Casdoor` can know where to send info back after authorization  
+
++ For `state`: you should fill this out with your `Application` name  
+
+The `Application` will speak to the user: *"Hey, now I need some resources and I need your permission to take these resources, you mind go to this `URL` and fill out your username and password for me?"*  
+
+With the correctly composed `URL`, your `Application` will make a user launch a request to this `URL`, and the `Authorization Request` is done.  
+
+### Step 2 (Authorization Grant) 
+
+This step is straightforward: the user is redirected to the `URL` composed above, and the user should see the login page from `Casdoor`.  
+By typing the correct username and credential into the login form, `Casdoor` now knows the identity of the user, and is about to send two keywords: `code` and `state` back to the callback URL set in Step 1.  
+
```

Comment:
Casdoor also provides third party login, please add the third party login case

Reply:
Sure